app/src/main/java/com/appharbr/example/app/max/MaxRewardedActivity.java

```diff
@@ -0,0 +1,121 @@
+package com.appharbr.example.app.max;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.appharbr.example.app.databinding.ActivityMaxRewardedBinding;
+import com.appharbr.sdk.engine.AdSdk;
+import com.appharbr.sdk.engine.AdStateResult;
+import com.appharbr.sdk.engine.AppHarbr;
+import com.appharbr.sdk.engine.listeners.AHListener;
+import com.applovin.mediation.MaxAd;
+import com.applovin.mediation.MaxError;
+import com.applovin.mediation.MaxReward;
+import com.applovin.mediation.MaxRewardedAdListener;
+import com.applovin.mediation.ads.MaxRewardedAd;
+import com.applovin.sdk.AppLovinSdk;
+
+import java.util.Arrays;
+
+public class MaxRewardedActivity extends AppCompatActivity {
+
+    private ActivityMaxRewardedBinding binding;
+    private MaxRewardedAd maxRewardedAd;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+	super.onCreate(savedInstanceState);
+
+	binding = ActivityMaxRewardedBinding.inflate(getLayoutInflater());
+	setContentView(binding.getRoot());
+
+	//Initialize AppLovinSdk
+	AppLovinSdk.getInstance(this).setMediationProvider("max");
+	AppLovinSdk.initializeSdk(this);
+
+	//Initialize max Rewarded Ad
+	maxRewardedAd = MaxRewardedAd.getInstance("YOUR_AD_UNIT_ID", this);
+
+	// The publisher will initiate once the listener wrapper and will use it when load the Max rewarded ad.
+	MaxRewardedAdListener ahWrapperListener = AppHarbr.addRewardedAd(AdSdk.MAX,
+		maxRewardedAd,
+		maxRewardedAdListener,
+		getLifecycle(),
+		ahListener);
+
+	//Set ahWrapperListener and load Ad
+	maxRewardedAd.setListener(ahWrapperListener);
+	maxRewardedAd.loadAd();
+    }
+
+    private final MaxRewardedAdListener maxRewardedAdListener = new MaxRewardedAdListener() {
+
+	@Override
+	public void onAdLoaded(MaxAd ad) {
+	    Log.d("LOG", "Max - onAdLoaded");
+	    binding.progressBar.setVisibility(View.GONE);
+	    if (maxRewardedAd.isReady()) {
+		checkAd();
+	    }
+	}
+
+	private void checkAd() {
+	    final AdStateResult interstitialState = AppHarbr.getInterstitialState(maxRewardedAd);
```

Comment:
Should use the public method AppHarbr.getRewardedState instant of getInterstitialState.